### PR TITLE
Warehouse picking feature integration guides.

### PR DESCRIPTION
**Pull request by abitskil**
* Epic: https://spryker.atlassian.net/browse/CC-20611
* Ticket: https://spryker.atlassian.net/browse/CC-26159

### CODE REVIEW

File: _includes/pbc/all/install-features/202304.0/install-the-order-management-feature.md [hunk 1]
```diff
@@ -0,0 +1,1334 @@
...
+```
+
+{% info_block warningBox "Verification" %}
+
+Make sure that the following modules have been installed:
+
+| MODULE                  | EXPECTED DIRECTORY                        |
+| --------------------- | --------------------------------------- |
+| OrderCustomReference    | vendor/spryker/order-custom-reference     |
+| OrderCustomReferenceGui | vendor/spryker/order-custom-reference-gui |
```
Comment (abitskil) [inline _includes/pbc/all/install-features/202304.0/install-the-order-management-feature.md:46]: * We have to add the BAPI module to the list.

File: _includes/pbc/all/install-features/202304.0/install-the-order-management-feature.md [hunk 2]
```diff
@@ -0,0 +1,1334 @@
...
+This document describes how to integrate the Order Management feature into a Spryker project.
+
+{% info_block warningBox "Included features" %}
+
+The following feature integration guide expects the basic feature to be in place. It only adds the following functionalities:
+
+- Order cancellation behavior
+- Show `display names` for order item states
+- Invoice generation
+- Custom order reference
```
Comment (abitskil) [inline _includes/pbc/all/install-features/202304.0/install-the-order-management-feature.md:11]: * We have to extend the guide with info about the SalesOrdersBackendApi module.

File: _includes/pbc/all/install-features/202304.0/install-the-cart-notes-feature.md [hunk 3]
```diff
@@ -0,0 +1,83 @@
+
+
+{% info_block errorBox %}
+
+This feature integration guide expects the basic feature to be in place.
+The current feature integration guide adds the following functionalities:
+* Translation TODO:
```
Comment (abitskil) [inline _includes/pbc/all/install-features/202304.0/install-the-cart-notes-feature.md:7]: * We have to fix the TODO in the document.

File: _includes/pbc/all/install-features/202304.0/install-the-order-management-feature.md [hunk 4]
```diff
@@ -0,0 +1,1334 @@
...
+| Mailing & Notifications | {{page.version}} |
+| Order Management        | {{page.version}} |
+| Persistent Cart         | {{page.version}} |
+
+### 1) Install the required modules using Composer
+
+Install the required modules:
+
+```bash
+composer require spryker-feature/order-management: "{{page.version}}r" --update-with-dependencies
```
Comment (DmitryLymarenko) [inline _includes/pbc/all/install-features/202304.0/install-the-order-management-feature.md:36]:
```suggestion
composer require spryker-feature/order-management: "{{page.version}}" --update-with-dependencies
```

File: _includes/pbc/all/install-features/202304.0/install-the-warehouse-picking-feature.md [hunk 5]
```diff
@@ -11,12 +11,13 @@ Follow the steps below to install the Warehouse picking feature core.
...
+| Order Management + Inventory Management      | {{site.version}} | [Order Management and Inventory Management feature](/docs/scos/dev/feature-integration-guides/{{site.version}}/install-the-order-management-and-inventory-management-feature.html) |
+| Shipment                                     | {{site.version}} | [Install the Shipment feature](/docs/scos/dev/feature-integration-guides/{{site.version}}/shipment-feature-integration.html)                                                       |
+| Push Notification                            | {{site.version}} | [Install the Push Notification feature](/docs/scos/dev/feature-integration-guides/{{site.version}}/install-the-push-notification-feature.md.html)                                  |
+| Spryker core backoofice (check the previous) |                  |                                                                                                                                                                                    |
```
Comment (abitskil) [inline _includes/pbc/all/install-features/202304.0/install-the-warehouse-picking-feature.md:20]: | Spryker Core Back Office | {{site.version}} | [Spryker Core Back Office feature integration](/docs/scos/dev/feature-integration-guides/{{site.version}}/spryker-core-back-office-feature-integration.html)

File: _includes/pbc/all/install-features/202304.0/install-the-warehouse-picking-feature.md [hunk 6]
```diff
@@ -243,6 +247,57 @@ Make sure that the OMS transition diagram shows the expected transitions:
...
+     * @return string
+     */
+    public function getPushNotificationProviderName(): string
+    {
+        return static::PUSH_NOTIFICATION_PROVIDER_NAME;
+    }
+}
+```
+
+4. Configure the `User` module to execute post save plugins: // TODO: check if this one is correct
```
Comment (abitskil) [inline _includes/pbc/all/install-features/202304.0/install-the-warehouse-picking-feature.md:280]: * This part has to be moved to the 'Spryker Core Back Office' integration guide.

File: _includes/pbc/all/install-features/202304.0/install-the-warehouse-picking-feature.md [hunk 7]
```diff
@@ -398,6 +458,115 @@ class OmsDependencyProvider extends SprykerOmsDependencyProvider
 }
 ```
 
+**src/Pyz/Glue/SalesOrdersBackendApi/SalesOrdersBackendApiDependencyProvider.php**
```
Comment (abitskil) [inline _includes/pbc/all/install-features/202304.0/install-the-warehouse-picking-feature.md:461]: * This is the part of the cart-note IG.

File: _includes/pbc/all/install-features/202304.0/install-the-packaging-units-feature.md [hunk 8]
```diff
@@ -0,0 +1,1063 @@
...
+
+Set up the following behaviors.
+
+#### Set up checkout workflow
+
+Enable the following behaviors by registering the plugins:
+
+| PLUGIN | SPECIFICATION | PREREQUISITES | NAMESPACE |
+| --- | --- | --- | --- |
+| AmountAvailabilityCartPreCheckPlugin | Validates if the given amount is available according to stock configuration during the cart change. | Expects the `amount` field to be set in `ItemTransfers` with packaging units. | Spryker\Zed\ProductPackagingUnit\Communication\Plugin\Cart |
```
Comment (abitskil) [inline _includes/pbc/all/install-features/202304.0/install-the-packaging-units-feature.md:551]: * We have to add the following missing plugin: PackagingUnitSplittableItemTransformerStrategyPlugin.

File: _includes/pbc/all/install-features/202304.0/install-the-packaging-units-feature.md [hunk 9]
```diff
@@ -0,0 +1,1063 @@
...
+| AmountAvailabilityCheckoutPreConditionPlugin | Validates if the given amount is available according to stock configuration during checkout. |Expects the `amount` and `amountLeadProduct` fields to be set in `ItemTransfers` with packaging units.  | Spryker\Zed\ProductPackagingUnit\Communication\Plugin\Checkout |
+| ProductPackagingUnitCartAddItemStrategyPlugin |Merges the `quantity` and `amount` fields changes into the cart for `ItemTransfers` with packaging units on the persistent cart add action.  | Expects the `amount` field to be set in `ItemTransfers` with packaging units. | Spryker\Zed\ProductPackagingUnit\Communication\Plugin\PersistentCart |
+| LeadProductReservationHandlerPlugin | Updates availability and reservation of a lead product for a given product packaging unit for reservation handler. |None  | Spryker\Zed\ProductPackagingUnit\Communication\Plugin\Reservation |
+| AmountLeadProductHydrateOrderPlugin | Hydrates the `leadProduct` field for Order read. | Expects the `items`, `amount`, and `amountSku` fields to be set in Order. | Spryker\Zed\ProductPackagingUnit\Communication\Plugin\Sales |
```
Comment (abitskil) [inline _includes/pbc/all/install-features/202304.0/install-the-packaging-units-feature.md:564]: * This is the deprecated plugin we have to remove it from the definition.

File: _includes/pbc/all/install-features/202304.0/install-the-packaging-units-feature.md [hunk 10]
```diff
@@ -0,0 +1,1063 @@
...
+| AmountSalesUnitHydrateOrderPlugin | Hydrates the `amountSalesUnit` field for Order read. | Expects the `items` field to be set in Order. | Spryker\Zed\ProductPackagingUnit\Communication\Plugin\Sales |
```
Comment (abitskil) [inline _includes/pbc/all/install-features/202304.0/install-the-packaging-units-feature.md:565]: This is the deprecated plugin we have to remove it from the definition (new one AmountSalesUnitOrderItemExpanderPlugin).

File: _includes/pbc/all/install-features/202304.0/install-the-packaging-units-feature.md [hunk 11]
```diff
@@ -0,0 +1,1063 @@
...
+{% endinfo_block %}
+
+{% info_block warningBox "Verification" %}
+
+Make sure that every order item from `SalesFacade::getOrderItems()` results contains packaging units data: `ItemTransfer.amountLeadProduct` and `ItemTransfer.amountSalesUnit` are set for the order items that have packaging units.
+
+{% endinfo_block %}
+
+
+## Install feature frontend
```
Comment (abitskil) [inline _includes/pbc/all/install-features/202304.0/install-the-packaging-units-feature.md:930]: * We have to add the missing plugin:
QuickOrderItemDefaultPackagingUnitExpanderPlugin

File: _includes/pbc/all/install-features/202304.0/install-the-packaging-units-feature.md [hunk 12]
```diff
@@ -0,0 +1,1098 @@
...
+```
+
+{% info_block warningBox "Verification" %}
+Make sure that the following modules have been installed:
+
+| MODULE                         | EXPECTED DIRECTORY                                |
+|--------------------------------|---------------------------------------------------|
+| ProductPackagingUnit           | vendor/spryker/product-packaging-unit             |
+| ProductPackagingUnitDataImport | vendor/spryker/product-packaging-unit-data-import |
+| ProductPackagingUnitStorage    | vendor/spryker/product-packaging-unit-storage     |
```
Comment (abitskil) [inline _includes/pbc/all/install-features/202304.0/install-the-packaging-units-feature.md:31]: We have to add the ProductPackagingUnitsBackendApi to the list.

File: _includes/pbc/all/install-features/202304.0/install-the-packaging-units-feature.md [hunk 13]
```diff
@@ -0,0 +1,1098 @@
...
+|--------------------------------------------------------------|--------------------------------------------------------------------------------------------------------------------------------------------------|-----------------------------------------------------------------------------------------------------------------------------|-------------------------------------------------------------------------|
+| AmountAvailabilityCartPreCheckPlugin                         | Validates if the given amount is available according to stock configuration during the cart change.                                              | Expects the `amount` field to be set in `ItemTransfers` with packaging units.                                               | Spryker\Zed\ProductPackagingUnit\Communication\Plugin\Cart              |
```
Comment (abitskil) [inline _includes/pbc/all/install-features/202304.0/install-the-packaging-units-feature.md:529]: We have to add  the ProductPackagingUnitApiPickingListItemsAttributesMapperPlugin.

File: _includes/pbc/all/install-features/202304.0/install-the-packaging-units-feature.md [hunk 14]
```diff
@@ -0,0 +1,1098 @@
...
+| spy_sales_order_item.amount_sku        | column | created |
+
+{% endinfo_block %}
+
+{% info_block warningBox "Verification" %}
+
+Make sure that the following changes in transfer objects have been applied:
+
+| TRANSFER                                     | TYPE     | EVENT   | PATH                                                                       |
+|----------------------------------------------|----------|---------|----------------------------------------------------------------------------|
```
Comment (abitskil) [inline _includes/pbc/all/install-features/202304.0/install-the-packaging-units-feature.md:135]: We have to extend the list based on the current state of the modules:
* ProductPackagingUnit
* ProductPackagingUnitsBackendApi

File: _includes/pbc/all/install-features/202304.0/install-the-packaging-units-feature.md [hunk 15]
```diff
@@ -0,0 +1,1131 @@
...
+| ProductPackagingUnitPickingListCollectionExpanderPlugin       | Expands `PickingListCollectionTransfer.pickingList.pickingListItem.orderItem` transfer objects with `amountSalesUnit` property.                                         |                                                                                                                                                                                                                 | Spryker\Zed\ProductPackagingUnit\Communication\Plugin\PickingList          |
+| ProductPackagingUnitApiPickingListItemsAttributesMapperPlugin | Maps amount sales unit from `PickingListItemTransfer.orderItem.amountSalesUnit` to `ApiPickingListItemsAttributesTransfer.orderItem.amountSalesUnit` transfer property. | Expects the `uuid` and `orderItem.amountSalesUnit.productMeasurementUnit` fields to be set in `PickingListItemTransfer`. Expects the `uuid` field to be set in `ApiPickingListItemsAttributesTransfer` as well. | Spryker\Glue\ProductPackagingUnitsBackendApi\Plugin\PickingListsBackendApi |                                                                         |
```
Comment (abitskil) [inline _includes/pbc/all/install-features/202304.0/install-the-packaging-units-feature.md:585]: * We have to add the code snippet for it.

**Review by abitskil — COMMENTED** [review]
Looks good from the technical side.

**Review by VadymSachenko — APPROVED** [review]

**Review by VadymSachenko — APPROVED** [review]

**Review by VadymSachenko — APPROVED** [review]

**Pull request merged**